api/type.rb

```diff
@@ -325,46 +325,38 @@ def field_name
 
     # Represents a reference to another resource
     class ResourceRef < Type
-      ALLOWED_WITHOUT_PROPERTY = [SelfLink::EXPORT_KEY].freeze
-
-      attr_reader :resource
-      attr_reader :imports
-
-      def out_type
-        resource_ref.out_name
-      end
+      attr_reader :resources
 
       def validate
         super
         @name = @resource if @name.nil?
         @description = "A reference to #{@resource} resource" \
           if @description.nil?
 
-        return if @__resource.nil? || @__resource.exclude || @exclude
+        @resources = @resources.map do |hash|
+          IndividualResourceRef.new(hash)
+        end
 
-        check_property :resource, ::String
-        check_property :imports, ::String
-        check_resource_ref_exists
-        check_resource_ref_property_exists
-      end
+        @resources.each do |p|
```

Comment:
This doesn't quite seem like the right place for this, can you add a comment or something explaining it? Should the resource's parent be the ResourceRef?

Reply:
Yeah, sounds good.

As for the resource's parent part, I'm following the pattern we established with NestedObjects + Arrays.